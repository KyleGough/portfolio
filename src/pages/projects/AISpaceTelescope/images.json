[
  { "src": "/img/ai-space-telescope1.jpg", "caption": "AI Space Telescope Website" }
]
