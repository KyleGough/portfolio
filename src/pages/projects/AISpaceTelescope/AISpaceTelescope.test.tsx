import { render } from '@testing-library/react';
import React from 'react';
import { BrowserRouter } from 'react-router-dom';

import { AISpaceTelescope } from './AISpaceTelescope';

describe('AISpaceTelescope component', () => {
  it('renders', async () => {
    const { container } = render(
      <BrowserRouter>
        <AISpaceTelescope />
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
