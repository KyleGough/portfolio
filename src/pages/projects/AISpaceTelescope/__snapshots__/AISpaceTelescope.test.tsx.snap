// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AISpaceTelescope component renders 1`] = `
<div>
  <div
    class="first:pt-24 container text-primary py-8 last:pb-16"
  >
    <h1
      class="project-title"
    >
      AI Space Telescope
    </h1>
    <p
      class="text-link-hover my-4"
    >
      <time
        datetime="2022-11"
      >
        November 2022
      </time>
    </p>
    <p
      class="mb-4 max-w-reading"
    >
      Gallery of science-fiction themed images generated using the amazing DALLÂ·E 2 API. These pictures are a hand-picked selection of my favourite generated images.
    </p>
    <ul
      class="project-list"
    >
      <li
        class="mb-4"
      >
        <span>
          Explore this project on 
        </span>
        <a
          class="text-link hover:text-link-hover focus:text-link-hover"
          href="https://github.com/KyleGough/ai-space-telescope"
          rel="noopener noreferrer"
          target="_blank"
        >
          GitHub
        </a>
      </li>
      <li
        class="mb-4"
      >
        <span>
          Explore this project live 
        </span>
        <a
          class="text-link hover:text-link-hover focus:text-link-hover"
          href="https://ai-space-telescope.com"
          rel="noopener noreferrer"
          target="_blank"
        >
          here
        </a>
      </li>
    </ul>
    <div
      class="flex flex-row flex-wrap items-center mt-8 gap-4"
    >
      <div
        class="text-chip bg-chip-light text-xs px-4 py-0.5 h-8 leading-chip rounded-2xl font-extrabold"
      >
        React
      </div>
      <div
        class="text-chip bg-chip-light text-xs px-4 py-0.5 h-8 leading-chip rounded-2xl font-extrabold"
      >
        Typescript
      </div>
    </div>
  </div>
  <hr
    class="m-8 h-px bg-divider"
  />
  <div
    class="first:pt-24 container text-primary py-8 last:pb-16"
  >
    <img
      alt="AI Space Telescope Website"
      class="border-2 border-white rounded-lg text-center items-center shadow mx-auto"
      src="/img/ai-space-telescope1.jpg"
    />
    <p
      class="mt-4 mb-2 text-sm text-center text-light font-thin"
    >
      AI Space Telescope Website
    </p>
  </div>
  <hr
    class="m-8 h-px bg-divider"
  />
  <div
    class="first:pt-24 container text-primary py-8 last:pb-16"
  >
    <h2
      class="project-header"
    >
      Implementation
    </h2>
    <p
      class="max-w-reading mb-4"
    >
      The images present on the website have been generated using OpenAI DALLE 2 API by providing space related prompts. Lots of different images have been generated from text prompts or reference images of previously generated images. Of these generated images only a few have been hand-selected to be shown on this website.
    </p>
    <p
      class="max-w-reading"
    >
      AI Space Telescope is a web-app developed using React, TailwindCSS, and TypeScript. Images are stored and served via Sirv, an image CDN which helps optimise images by serving images in the best format and size. The website is deployed on Vercel, and uses Cloudflare services for DNS, and security.
    </p>
  </div>
  <hr
    class="m-8 h-px bg-divider"
  />
  <div
    class="first:pt-24 container text-primary py-8 last:pb-16"
  >
    <h2
      class="text-3xl mb-12"
    >
      Screenshots
    </h2>
    <div
      class="grid grid-cols-12 gap-4"
    >
      <figure
        class="col-span-12 md:col-span-6 lg:col-span-4"
      >
        <img
          alt="AI Space Telescope Website"
          class="cursor-pointer border-2 border-link hover:border-link-hover"
          loading="lazy"
          src="/img/ai-space-telescope1.jpg"
        />
        <figcaption
          class="my-2 text-sm text-center text-light font-thin"
        >
          AI Space Telescope Website
        </figcaption>
      </figure>
    </div>
  </div>
  <hr
    class="m-8 h-px bg-divider"
  />
  <div
    class="first:pt-24 container text-primary py-8 last:pb-16"
  >
    <div
      class="flex justify-between"
    >
      <a
        class="flex items-center fill-link text-link hover:fill-link-hover hover:text-link-hover"
        href="/projects/bookmark-labeller"
      >
        <svg
          class="w-6 h-6"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 0h24v24H0V0z"
            fill="none"
            opacity=".87"
          />
          <path
            d="M17.51 3.87L15.73 2.1 5.84 12l9.9 9.9 1.77-1.77L9.38 12l8.13-8.13z"
          />
        </svg>
        <div
          class="text-center ml-4"
        >
          <p
            class="text-sm"
          >
            Previous Project
          </p>
          <p>
            Bookmark Labeller
          </p>
        </div>
      </a>
      <a
        class="flex items-center fill-disabled text-disabled cursor-not-allowed"
        href="/"
      >
        <div
          class="text-center mr-4"
        >
          <p
            class="text-sm"
          >
            Next Project
          </p>
          <p>
            -
          </p>
        </div>
        <svg
          class="w-6 h-6"
          enable-background="new 0 0 24 24"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g>
            <path
              d="M0,0h24v24H0V0z"
              fill="none"
            />
          </g>
          <g>
            <polygon
              points="6.23,20.23 8,22 18,12 8,2 6.23,3.77 14.46,12"
            />
          </g>
        </svg>
      </a>
    </div>
  </div>
</div>
`;
